### ISSUE TITLE
UPS not found even though device is there

### ISSUE THREAD

**Issue by Vieplis** [fajfer/gcups#13]
Tried running gcups in my environment with UPS01LCD version, but kind of the same issue as here: https://github.com/fajfer/gcups/issues/12
Docker container starts, but throws error and does not see the UPS even though it seems to be connected and detected by the OS itself.

```
x@docker-ubuntu:~$ lsusb
Bus 003 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 002 Device 002: ID 0001:0000 Fry's Electronics MEC0003
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 0627:0001 Adomax Technology Co., Ltd QEMU USB Tablet
Bus 001 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
```

Should be 002/002. Then starting the container with:
`sudo docker run --device=/dev/bus/usb/002/002 -p 0.0.0.0:8080:8080 ghcr.io/fajfer/gcups:1.1.7
`
 but getting the same error as in FAQ when UPS is not detected (and GUI also shows - "The device is not connected"):
```
[106:1020/141952.677608:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -101
/opt/gcups/gcups[538]: ../../third_party/electron_node/src/node_api.cc:1332:napi_status napi_release_threadsafe_function(napi_threadsafe_function, napi_threadsafe_function_release_mode): Assertion `(func) != nullptr' failed.
[106:1020/141953.973916:ERROR:ssl_client_socket_impl.cc(992)] handshake failed; returned -1, SSL error code 1, net_error -101
```

Setup is Celeron based minipc with Proxmox running Ubuntu 22.04.5 LTS with Docker 27.3.1. USB device is added to VM via passthrough of the port. Tried rebooting Ubuntu to re-initialize USB devices, but didn't help.
Not sure if that is actually an issue, could be something to do with my environment, but any ideas?

**Comment by fajfer** [comment]
Hey @Vieplis 

We were debugging this on Matrix and some people pointed out that docker started working once they installed and removed `gcups` apt package. We thought it could be the dependencies but I wasn't able to replicate that on a VM.

I didn't check it at first but now I'm 99% sure it's the scripts related to this package, the apt package contains additional configuration and it's not applied with just running the docker.

The scripts are
```bash
install() {
    :
    echo 'before-install'
tee /etc/udev/rules.d/10-gcups.rules > /dev/null <<EOT
SUBSYSTEM=="input", GROUP="input", MODE="0666"
SUBSYSTEM=="usb", ATTRS{idVendor}=="0001", SYMLINK+="ups", ATTRS{idProduct}=="0000", MODE:="666", GROUP="plugdev"
KERNEL=="hidraw*", ATTRS{idVendor}=="0001", ATTRS{idProduct}=="0000", MODE="0666", GROUP="plugdev"
EOT

tee /etc/udev/rules.d/10-gcups17.rules > /dev/null <<EOT
SUBSYSTEM=="input", GROUP="input", MODE="0666"
SUBSYSTEM=="tty", ATTRS{idVendor}=="067b", SYMLINK+="ttyups", ATTRS{idProduct}=="2303", MODE:="666", GROUP="plugdev"
EOT
    udevadm control --reload-rules
    common
}
```
and
```bash
common() {
    :
tee "$appDir/resources/dist.json" > /dev/null <<-EOT
{
    "updater": true,
    "target": "deb"
}
EOT

    ln -sf "$appDir/$executable" "/usr/bin/$executable"
    mkdir -p -m 777 "$appDir/db"
    mkdir -p -m 777 "$appDir/log"
    chmod 4755 "$appDir/chrome-sandbox" || true
    update-mime-database /usr/share/mime || true
    update-desktop-database /usr/share/applications || true
}
```
The easiest way to run this is to run `apt install gcups`, I've tinkered with it but currently I'm not sure how to approach this properly at this point.

**Comment by Vieplis** [comment]
Yes, indeed - installed gcups and that solved the problem. I'd say it is completely fine for me, thank your work and this tool!

**Comment by fajfer** [comment]
I have since fixed it by recommending users to run [udev.sh](https://github.com/fajfer/gcups/blob/main/udev.sh) to setup udev rules prior to running the container

**Comment by lbFakier** [comment]
not working on my setup,
on 'http://192.168.1.5:8080/en' page i can see 'Device is not connected' info,
docker log:
`* system message bus already started; not starting.

GCUPS running on port: 8080

Default webUI password: gcups123

Running gcups [30:0109/223433.075469:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

v1.1.7

[60:0109/223433.722578:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

[60:0109/223434.439649:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

[60:0109/223434.439689:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

[60:0109/223434.505248:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

[60:0109/223434.505299:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

[60:0109/223434.505325:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: Connection refused

/opt/gcups/gcups[115]: ../../third_party/electron_node/src/node_api.cc:1332:napi_status napi_release_threadsafe_function(napi_threadsafe_function, napi_threadsafe_function_release_mode): Assertion `(func) != nullptr' failed.`

tried 'apt install gcups' -> 'gcups is already the newest version (1.1.7).'

tried ' bash udev.sh' -> nothing new... still not working

lsusb
Bus 001 Device 005: ID 0001:0000 Fry's Electronics MEC0003

devices:
      - /dev/bus/usb/001/005:/dev/bus/usb/001/005

**Comment by fajfer** [comment]
@lbFakier looks good, save for the errors. Have you tried running it with just the system package instead of container? Does it work for you that way? It kinda looks like there's something wrong with the usb, maybe it's the cable

**Comment by lbFakier** [comment]
looks really strange, but it started working right now, 
I've unplugged the USB cable, plugged it again, 
usb device increased from 5 to 6
'Bus 001 Device 006: ID 0001:0000 Fry's Electronics MEC0003'

I've edited the stack (005 -> 006), updated it, and now it is working.

Thank you, for your help.
